Section 4.2

aleno4ka1998 opened the pull request.

aleno4ka1998 committed:
Делаем срезы

Выполните разбор строки, согласно приведенным правилам.

Входные данные
Дана строка.

Выходные данные
Сначала выведите третий символ этой строки.

Во второй строке выведите предпоследний символ этой строки.

В третьей строке выведите первые пять символов этой строки.

В четвертой строке выведите всю строку, кроме последних двух символов.

В пятой строке выведите все символы с четными индексами (считая, что индексация начинается с 0, поэтому символы выводятся начиная с первого).

В шестой строке выведите все символы с нечетными индексами, то есть начиная со второго символа строки.

В седьмой строке выведите все символы в обратном порядке.

В восьмой строке выведите все символы строки через один в обратном порядке, начиная с последнего.

В девятой строке выведите длину данной строки.

aleno4ka1998 committed:
Количество слов

Дана строка, состоящая из слов, разделенных пробелами. Определите, сколько в ней слов. Используйте для решения задачи метод count.

Входные данные
Вводится строка.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Две половинки

Дана строка. Разрежьте ее на две равные части (если длина строки — четная, а если длина строки нечетная, то длина первой части должна быть на один символ больше). Переставьте эти две части местами, результат запишите в новую строку и выведите на экран.

При решении этой задачи нельзя пользоваться инструкцией if.

Входные данные
Вводится строка.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Переставить два слова

Дана строка, состоящая ровно из двух слов, разделенных пробелом. Переставьте эти слова местами. Результат запишите в строку и выведите получившуюся строку.

При решении этой задачи нельзя пользоваться циклами и инструкцией if.

thelightningseas commented on section4.2/two halves.py:
```diff
@@ -0,0 +1,2 @@
+s = str(input())
```
не надо делать ```str(input())```. так приходилось делать когда считывали число (из строки в инт например хотим конвертнуть), а тут у нас и так строка будет

aleno4ka1998 committed:
Первое и последнее вхождение

Дана строка. Если в этой строке буква f встречается только один раз, выведите её индекс. Если она встречается два и более раз, выведите индекс её первого и последнего появления. Если буква f в данной строке не встречается, ничего не выводите.

При решении этой задачи нельзя использовать метод count и циклы.

Входные данные
Вводится строка.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Второе вхождение

Дана строка. Найдите в этой строке второе вхождение буквы f, и выведите индекс этого вхождения. Если буква f в данной строке встречается только один раз, выведите число -1, а если не встречается ни разу, выведите число -2.

При решении этой задачи нельзя использовать метод count.

Входные данные
Вводится строка.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Удаление фрагмента

Дана строка, в которой буква h встречается минимум два раза. Удалите из этой строки первое и последнее вхождение буквы h, а также все символы, находящиеся между ними.

Входные данные
Вводится строка.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Обращение фрагмента

Дана строка, в которой буква h встречается как минимум два раза. Разверните последовательность символов, заключенную между первым и последнием появлением буквы h, в противоположном порядке.

Входные данные
Вводится строка.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Дублирование фрагмента

Дана строка, в которой буква h встречается как минимум два раза. Повторите последовательность символов, заключенную между первым и последнием появлением буквы h два раза, сами буквы h повторять не надо.

Входные данные
Вводится строка.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Замена подстроки

Дана строка. Замените в этой строке все цифры 1 на слово one.

Входные данные
Вводится строка.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Удаление символа

Дана строка. Удалите из этой строки все символы @.

Входные данные
Вводится строка.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Замена внутри фрагмента

Дана строка. Замените в этой строке все появления буквы h на букву H, кроме первого и последнего вхождения.

Входные данные
Вводится строка.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Вставка символов

Дана строка. Получите новую строку, вставив между двумя символами исходной строки символ *. Выведите полученную строку.

Входные данные
Вводится строка.

Выходные данные
Выведите ответ на задачу.

aleno4ka1998 committed:
Удалить каждый третий символ

Дана строка. Удалите из нее все символы, чьи индексы делятся на 3.

Входные данные
Вводится строка.

Выходные данные
Выведите ответ на задачу.

thelightningseas commented on section4.2/first and last occurrence.py:
```diff
@@ -0,0 +1,8 @@
+s = str(input())
```
а если эта строка длины 40 миллионов символов, хорошо ли будет работать такой код? как тут можно соптимизировать, чтобы работало быстрее?

thelightningseas replied:
уже лучше но можно раза в 3 еще быстрее

thelightningseas replied:
да теперь супер

aleno4ka1998 committed:
(Исправ) Первое и последнее вхождение

Вроде полегче код стал

aleno4ka1998 committed:
upd Первое и последнее вхождение

Если найти индексы f один раз и работать дальше только с ними, то скорость обработки повысится
